sdk/servicebus/azure-messaging-servicebus-track1-perf/src/main/java/com/microsoft/azure/messaging/servicebus/perf/ReceiveAndDeleteMessageTest.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.messaging.servicebus.perf;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.azure.messaging.servicebus.perf.core.ServiceBusStressOptions;
+import com.microsoft.azure.messaging.servicebus.perf.core.ServiceTest;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.ReceiveMode;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Performance test.
+ */
+public class ReceiveAndDeleteMessageTest extends ServiceTest<ServiceBusStressOptions> {
+    private final ClientLogger logger = new ClientLogger(ReceiveAndDeleteMessageTest.class);
+    private final ServiceBusStressOptions options;
+
+    /**
+     * Creates test object
+     * @param options to set performance test options.
+     */
+    public ReceiveAndDeleteMessageTest(ServiceBusStressOptions options) {
+        super(options, ReceiveMode.RECEIVEANDDELETE);
+        this.options = options;
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        return Mono.fromFuture(CompletableFuture.allOf(sender.closeAsync(), receiver.closeAsync()))
+            .then(super.globalCleanupAsync());
+    }
+
+    private Mono<Void> sendMessage() {
+        return Mono.fromCallable(() -> {
+            int total = options.getParallel() * options.getMessagesToSend() * TOTAL_MESSAGE_MULTIPLIER;
+
+            List<Message> messages = new ArrayList<>();
+            for (int i = 0; i < total; ++i) {
+                Message message = new Message(CONTENTS);
+                message.setMessageId(UUID.randomUUID().toString());
+                messages.add(message);
+            }
+
+            return sender.sendBatchAsync(messages).get();
```

Comment:
This still has a blocking call in an async method. Here's what I was thinking of. Usually, you want to push the blocking synchronous call as late as possible. There are probably ,other instances of this blocking call in the perf track 1 tests. I didn't check for other instances.

```java

    private Mono<Void> sendMessage() {
        int total = options.getParallel() * options.getMessagesToSend() * TOTAL_MESSAGE_MULTIPLIER;

        List<Message> messages = new ArrayList<>();
        for (int i = 0; i < total; ++i) {
            Message message = new Message(CONTENTS);
            message.setMessageId(UUID.randomUUID().toString());
            messages.add(message);
        }

        return Mono.fromFuture(sender.sendBatchAsync(messages));
    }
```